component-data/brave-viewer/test.js

```diff
@@ -0,0 +1,17 @@
+new Promise(resolve => {
+  const mutationCallback = (mutationList, observer) => {
+    const finish = () => {
+      observer.disconnect();
+      resolve(true);
+    }
+    for (const mutation of mutationList) {
+      if (mutation.type === 'characterData' && mutation.target.textContent === "You're offline. Check your connection.") {
+        finish();
+      }
+    }
```

Comment:
note: this file is an _extremely_ rough draft of the detection logic, and will need to be improved either in this PR or in followups